### PR TITLE
Readability redundancy improvements

### PR DESCRIPTION
**Pull request by zlangner**
refactor: split extension class into 2 files

Split extension class into 2 files, one for just `double` the other for just `float` (makes it easier to tell that they are nearly identical).

test: updated tests to be less redundant

Updated tests to make use of XUnit MemberData so there is less redundancy in test code while making multiple use of the test scenarios. There is a set of test scenarios for:
  - Equals (used by EqualsTo, IsGreaterThanOrEquals, and IsLessThanOrEquals)
  - Not Equals (used by EqualsTo)
  - Zero (used by IsZero)
  - Comparison (Used by IsGreaterThan, IsGreaterThanOrEquals, IsLessThan, and IsLessThanOrEquals)

### CODE REVIEW

**Comment by zlangner** [comment]
@Commit-Lint review

**Comment by zlangner** [comment]
I'm not sure what commit-lint's issue is with the description on this PR.

**Comment by kerimbal** [comment]
> I'm not sure what commit-lint's issue is with the description on this PR.

Commit messages must follow the conventional commits spec **in order to deploy the packge with the correct version**. I see that you have edited the PR description accordingly. However, the PR description does not matter. It is the commit message that matters.

File: test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+using FluentAssertions;
+using Xunit;
+
+namespace ComparisonExtensionsTests
+{
+    public class DoubleComparisonExtensionsTest
+    {
+        public record DoubleEqualityScenario(double Left, double Right, string Name = null)
+        {
+            public override string ToString() => this.Name == null
```
Comment (kerimbal) [inline test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs]: We can use the null coalescing expression here. There are more of these. Please, do this for all of them.

Suggested code:
```CSharp
public override string ToString() => this.Name ?? $"left = {this.Left}, right = {this.Right}";
```

File: test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+            {
+                yield return new(0.1d, 0.1d);
+                yield return new(-0.1d, -0.1d);
+                yield return new(0.000000001d, 0.000000001d);
+                yield return new(double.PositiveInfinity, double.PositiveInfinity);
+                yield return new(double.NegativeInfinity, double.NegativeInfinity);
+                yield return new(double.Epsilon, 0.0d);
+                yield return new(0.0d, double.Epsilon);
+
+                yield return new((double)0.1 + (double)0.2 + (double)0.7, 1, "works well with integers");
```
Comment (kerimbal) [inline test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs]: casting to `double` is redundant here.

File: test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+            }
+        }
+
+        public static IEnumerable<object[]> ZeroTestCasesData => ZeroTestCases.Select(testCase => new object[] { testCase });
+        private static IEnumerable<DoubleZeroScenario> ZeroTestCases
+        {
+            get
+            {
+                // these are zero
+                yield return new((double)0.0, true);
```
Comment (kerimbal) [inline test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs]: casting to `double` is redundant here.

File: test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+                yield return new(double.Epsilon, true, "epsilon");
+                yield return new((double)0.3 - (double)0.2 - (double)0.1, true, "added doubles");
```
Comment (kerimbal) [inline test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs]: casting to `double` is redundant here.

**Review by kerimbal — CHANGES_REQUESTED** [review]

File: test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs (hunk 2 above)
Comment (zlangner) [reply test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs]: it is, I was trying to leave it how the original test was but it's updated now

File: test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs (hunk 1 above)
Comment (zlangner) [reply test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs]: updated

File: test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs (hunk 3 above)
Comment (zlangner) [reply test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs]: updated

File: test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs (hunk 4 above)
Comment (zlangner) [reply test/ComparisonExtensionsTests/DoubleComparisonExtensionsTest.cs]: updated

**Comment by zlangner** [comment]
I think I have all the commits reworded now.

Since all I'm really doing here is changing tests, what do you think about adding netstandard1.0 so this can be used by more than just net5.0? I'm happy to add that in this or a different PR. That would be a good reason to push a new version to nuget.

**Comment by zlangner** [comment]
apparently there is still an issue with these commit messages. I've read https://www.conventionalcommits.org/en/v1.0.0/ and the commit messages I have on the 2 commits should fit the convention. I'd appreciate any help you can give for why they don't. I would be certainly be nice for this lint tool to give feedback about what's wrong.

**Comment by kerimbal** [comment]
> I think I have all the commits reworded now.
> 
> Since all I'm really doing here is changing tests, what do you think about adding netstandard1.0 so this can be used by more than just net5.0? I'm happy to add that in this or a different PR. That would be a good reason to push a new version to nuget.

I would be glad it if you do so. Thanks.

**Pull request merged**